### DISCUSSION TITLE
Is it possible to disable liquid for markdown parsing

### DISCUSSION THREAD

**Discussion by nabeelvalley** [11ty/eleventy#2148] [Q&A]
I've got a bunch of markdown files that contain code blocks with content like `{{  myVariable }}`, especially the case of React code blocks

I am having an issue trying to replace `{{` in a custom markdown converter but it seems like the braces (and their content) are being replaced before the file is sent to my markdoown converter - I assume as part of the liquid parsing

Is there any way to stop this behavior or even totally diable liquid for markdown files?

**Comment by pdehaan** [comment]
Depending on your site, you can wrap code in a `{% raw %}` block to have it avoid parsing Liquid/Nunjucks tags.
- https://liquidjs.com/tags/raw.html
- https://mozilla.github.io/nunjucks/templating.html#raw

But you're trying to stop ALL liquid/nunjucks parsing in Markdown files (not just specific blocks of code)?

**Reply by pdehaan** [reply]
Maybe https://www.11ty.dev/docs/config/#default-template-engine-for-markdown-files (apparently you can set `markdownTemplateEngine: false` in your .eleventy.js config file).

Or try https://www.11ty.dev/docs/data-configuration/#configure-your-templates, setting `templateEngineOverride: md` in your .md template's frontmatter.

---

```
---
title: Hello!
templateEngineOverride: md
---

## {{ title }}

{% eleventyVersion %}
```

Where my .eleventy.js file looks like this:

```js
const eleventyPkg = require("@11ty/eleventy/package.json");

module.exports = (eleventyConfig) => {
  eleventyConfig.addShortcode("eleventyVersion", () => eleventyPkg.version);

  return {
    dir: {
      input: "src",
      output: "www"
    },
    // markdownTemplateEngine: false,
    markdownTemplateEngine: "liquid",
  };
};
```

And the generated HTML output looks like:

```html
<h2>{{ title }}</h2>
<p>{% eleventyVersion %}</p>
```

**Reply by pdehaan** [reply]
So I guess the long-winded answer was:
- use `markdownTemplateEngine: false` in .eleventy.js config file if you want to disable Liquid-in-Markdown parsing globally.
- use `templateEngineOverride: md` frontmatter if you want to disable LiquidJS parsing on a per-template basis.
- use `{% raw %}...{% endraw %}` wrapper if you want to prevent LiquidJS (or Nunjucks) from parsing specific blocks/tags in a file.

**Reply by pdehaan** [reply]
One more idea, for future travellers... This worked in @11ty/eleventy@beta (but not v0.12.x):

```js
// .eleventy.js
const eleventyVersion = require("@11ty/eleventy/package.json").version;
const liquidVersion = require("liquidjs/package.json").version;

module.exports = (eleventyConfig) => {
  eleventyConfig.addShortcode("eleventyVersion", () => eleventyVersion);
  eleventyConfig.addShortcode("liquidVersion", () => liquidVersion);
  
  eleventyConfig.setLiquidOptions({
    outputDelimiterLeft: "[[",
    outputDelimiterRight: "]]",
    tagDelimiterLeft: "[%",
    tagDelimiterRight: "%]",
  });

  return {
    dir: {
      input: "src",
      output: "www"
    },
  };
};
```

And then my Markdown template looks like this:

```md
---
title: Hello!
---

# [[ title ]]
eleventy: [% eleventyVersion %]
liquidjs: [% liquidVersion %]

---

# {{ title }}
{% eleventyVersion %}
```

And my output looks like:

```html
<h1>Hello!</h1>

<p>eleventy: 1.0.0-beta.8
liquidjs: 9.31.0</p>

<hr>

<h1>{{ title }}</h1>
<p>{% eleventyVersion %}</p>
```

You can see in the output that the overridden delimiters worked and the `{{ title }}` was untouched, so maybe it's possible to prerender Markdown w/ LiquidJS and still leave some tags for React, if that's your thing.
